test/integration/client-side-encryption/client_side_encryption.prose.18.azure_kms_mock_server.test.ts

```diff
@@ -0,0 +1,156 @@
+import { expect } from 'chai';
+import { get } from 'http';
+
+import { Document } from '../../mongodb';
+
+const BASE_URL = new URL(`http://127.0.0.1:8080/metadata/identity/oauth2/token`);
+
+async function mockServerIsSetup() {
+  const url = (() => {
+    const copiedURL = new URL(BASE_URL);
+
+    // minimum configuration for the mock server not to throw an error when responding.
+    copiedURL.searchParams.append('api-version', '2018-02-01');
+    copiedURL.searchParams.append('resource', 'https://vault.azure.net');
+    return copiedURL;
+  })();
+  return new Promise<void>((resolve, reject) => {
+    get(url, res => {
+      if (res.statusCode === 200) {
+        return resolve();
+      }
+      return reject('server not running');
+    })
+      .on('error', error => reject(error))
+      .end();
+  });
+}
+
+class KMSRequestOptions {
+  url: URL = BASE_URL;
+  headers: Document;
+  constructor(testCase?: 'empty-json' | 'bad-json' | '404' | '500' | 'slow') {
+    this.headers =
+      testCase != null
+        ? {
+            'X-MongoDB-HTTP-TestParams': `case=${testCase}`
+          }
+        : {};
+  }
+}
+
+context('Azure KMS Mock Server Tests', function () {
+  let fetchAzureKMSToken: (options: {
+    url: URL;
+    headers: Document;
+  }) => Promise<{ accessToken: string }>;
+  let KMSRequestFailedError: Error;
+
+  const AZURE_KMS_TEST_EXPORTS = '___azureKMSProseTestExports';
+  beforeEach(async function () {
+    try {
+      await mockServerIsSetup();
+    } catch {
+      this.currentTest.skipReason = 'Test requires mock azure identity endpoint to be running.';
+      this.test?.skip();
```

Comment:
Is there a concern that the server might be down within the CI run that we need it to be up? should this throw still if one of the AZURE env variables is defined?